src/components/permissions.tsx

```diff
@@ -28,7 +28,7 @@ const assets = {
   }
 }
 
-class Permissions extends React.Component {
+class Permissions extends React.Component<any, any> {
```

Comment:
For my own benefit, could you explain what these two types would be expected to be when we're not type-erasing them?

Reply:
Short answer: `<props, state>`

Long answer: If I were actually adding types and not just trying to avoid changes as much as possible for this initial PR,  then I might create a specific type for props and state then pass them in for those generic arguments like this: 

```ts
type ComponentProps = {
  propA: string,
  propB: number,
  propC: string[]
}

type ComponentState = {
  stateA: string,
  stateB: {
	property: string
  }
}

class Component extends React.Component<ComponentProps, ComponentState> {
...
```

But this syntax is only for class-based components, which are no longer preferred generally; though there are still use cases for them. Eventually, I would expect most (if not all) the class components to be refactored out of this project.

Functional components still use a prop type, but state types are handled differently.